src/github.com/stellar/gateway/listener/payment_listener_test.go

```diff
@@ -144,6 +144,62 @@ func TestPaymentListener(t *testing.T) {
 			})
 		})
 
+		Convey("When payment is XLM(no XLM asset in config)", func() {
+			operation.Type = "payment"
+			operation.From = "GBL27BKG2JSDU6KQ5YJKCDWTVIU24VTG4PLB63SF4K2DBZS5XZMWRPVU"
+			operation.To = "GATKP6ZQM5CSLECPMTAC5226PE367QALCPM6AFHTSULPPZMT62OOPMQB"
+			operation.AssetCode = ""
+			operation.AssetIssuer = ""
+			operation.AssetType = "native"
+			dbPayment.Status = "Asset not allowed"
+			mockEntityManager.On("Persist", &dbPayment).Return(nil).Once()
+			mockRepository.On("GetReceivedPaymentByOperationID", int64(1)).Return(nil, nil).Once()
+
+
+			Convey("it should save the status", func() {
+				err := paymentListener.onPayment(operation)
+				assert.Nil(t, err)
+				mockEntityManager.AssertExpectations(t)
+			})
+		})
+
+
+		Convey("When payment is XLM (XLM asset in config)", func() {
+			operation.Type = "payment"
+			operation.To = "GATKP6ZQM5CSLECPMTAC5226PE367QALCPM6AFHTSULPPZMT62OOPMQB"
+			operation.AssetCode = ""
+			operation.AssetIssuer = ""
+			operation.AssetType = "native"
+			dbPayment.Status = "Success"
+			operation.Memo.Type = "book"
+			operation.Memo.Value = "testing"
+			config.Assets[1].Code = "XLM"
+			config.Assets[1].Issuer = ""
```

Comment:
You should reset these values somewhere. Otherwise tests following this one will use updated config. `operation` and `dbPayment` are reset in the main `Convey("PaymentListener"` function.